crates/iota-genesis-builder/src/stardust/process_outputs.rs

```diff
@@ -142,67 +348,106 @@ impl<I: Iterator<Item = Result<(OutputHeader, Output)>>> Iterator for MergingIte
                 Ok((output_header_with_balance.output_header, basic.into()))
             })
     }
+
+    fn print_metrics(&self) {
+        if self.with_metrics {
+            info!(
+                "Number of UnlockedVestingOutputFilter modified outputs: {}",
+                self.modified_outputs.len()
+            );
+            info!(
+                "Number of UnlockedVestingOutputFilter outcome outputs: {}",
+                self.number_of_outcome_outputs
+            );
+        }
+    }
 }
 
-fn scale_output_amount_for_iota(output: &mut Output) -> Result<()> {
-    *output = match output {
-        Output::Basic(ref basic_output) => {
-            // Update amount
-            let mut builder = BasicOutputBuilder::from(basic_output)
-                .with_amount(scale_amount_for_iota(basic_output.amount())?);
+/// Filter that looks for basic outputs having a tag being the Participation Tag
+/// and removes all features from the basic output. Operates in place during the
+/// iteration.
+#[derive(Default)]
+struct VotingOutputFilter {
+    /// Outputs where features were removed
+    modified_outputs: Vec<OutputId>,
+    /// Flag to enable metrics
+    with_metrics: bool,
+}
 
-            // Update amount in potential storage deposit return unlock condition
-            if let Some(sdr_uc) = basic_output
-                .unlock_conditions()
-                .get(StorageDepositReturnUnlockCondition::KIND)
-            {
-                let sdr_uc = sdr_uc.as_storage_deposit_return();
-                builder = builder.replace_unlock_condition(
-                    StorageDepositReturnUnlockCondition::new(
-                        sdr_uc.return_address(),
-                        scale_amount_for_iota(sdr_uc.amount())?,
-                        u64::MAX,
-                    )
-                    .unwrap(),
-                );
-            };
+impl VotingOutputFilter {
+    fn new(with_metrics: bool) -> Self {
+        Self {
+            modified_outputs: vec![],
+            with_metrics,
+        }
+    }
+}
+
+impl HornetFilter<Result<(OutputHeader, Output)>> for VotingOutputFilter {
+    fn process(
+        &mut self,
+        output: Result<(OutputHeader, Output)>,
+    ) -> Option<Result<(OutputHeader, Output)>> {
+        // continue filtering
+        Some(output.map(|(header, inner)| {
+            let output_id = header.output_id();
+            (
+                header,
+                if is_voting_output(&inner) {
+                    if self.with_metrics {
+                        self.modified_outputs.push(output_id);
+                    }
+                    // replace the inner output
+                    BasicOutputBuilder::from(inner.as_basic())
+                        .clear_features()
```

Comment:
why do we have this special handling for the participation, and why do we simply clear all features here?

Reply:
@miker83z explained the reason in a call. Only then the outputs can be migrated to coin objects, and this is the intention of that change.

Reply:
@muXxer just to be on the safe side, with this commit [0e5a810](https://github.com/iotaledger/iota/pull/4513/commits/0e5a810461824b964bb6faf23ae65fefd10e82e3) sender and issuer features are kept.